[saraozn] iP

saraozn opened the pull request:
### NicoleGalaxyProFlipFlop

> "Life is uncertain. Eat dessert first." - Ernestine Ulmer ([source](https://pos.toasttab.com/blog/on-the-line/food-quotes-and-captions))

 Frees your mind of having to remember things you need to do, all you need to remember is eat . It's,

- text-based
- easy to learn
- ~FAST~ godspeed

All you need to do is,

1. download it from [here](https://nus-cs2103-ay2324s1.github.io/website/schedule/week4/project.html).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 
5. Eat 

And it is FREE!

Features:

- [ ]  Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)

Here's the main method:
```
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

saraozn committed:
feat: Rename, Greet, Exit

saraozn committed:
feat: Echo

saraozn committed:
feat: Add, List

saraozn committed:
feat: Mark as Done

saraozn committed:
feat: ToDos,Events, Deadlines

saraozn committed:
feat: Automated Text UI Testing

saraozn committed:
feat: Handle Errors

saraozn committed:
feat: Delete

saraozn committed:
feat: Save

saraozn committed:
no message

saraozn committed:
Merge branch 'branch-Level-7'

saraozn committed:
feat: Dates and Times

saraozn committed:
Merge branch 'branch-Level-8'

saraozn committed:
added new class Storage, and moved methods loadTasks and saveTasks into
the new class

saraozn committed:
feat: OOP

saraozn committed:
Merge branch 'branch-A-MoreOOP'

saraozn committed:
added packages

saraozn committed:
Merge branch 'add-gradle-support'

saraozn committed:
added parserTest class to test the method parse in Parser

saraozn committed:
added storageTest class to test the methods load and save

saraozn committed:
added javaDocs to most of the classes

saraozn committed:
Complied with coding standard

saraozn committed:
added findTask to find a task by searching a keyword

saraozn committed:
Merge branch 'branch-A-JavaDoc'

saraozn committed:
Merge branch 'branch-A-CodingStandard'

# Conflicts:
#	src/main/java/duke/Duke.java
#	src/main/java/duke/Parser.java
#	src/main/java/duke/TaskList.java
#	src/main/java/duke/Taskparser.java
#	src/main/java/duke/Todo.java
#	src/main/java/duke/Ui.java

saraozn committed:
Merge branch 'branch-Level-9'

saraozn committed:
added test class for TaskList tot test deleteTask and addTask methods

saraozn committed:
Resolved conflicts for JUnit and Master branch

cxo05 commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,43 @@
+    public ArrayList<Task> load() throws FileNotFoundException {
+        ArrayList<Task> dukeList = new ArrayList<>();
+        File file = new File(filepath);
+        if (file.exists()) {
```
If the file does not exist, create the file directory and the file itself. The method will then not need a `FileNotFoundException`. 

```java
File file = new File(filepath);
if (!taskFile.exists()) {
  taskFile.mkdirs(); //Make the directory if it doesn't exist
}

if (taskFile.createNewFile()) { //Returns false if the file exists else create the file and return true
  System.out.println("File created: " + taskFile.getName());
}
```

cxo05 commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,88 @@
+public class TaskList {
+
+    ArrayList<Task> list;
+    int count;
```
The `ArrayList` object has the .size() property which will remove the need for `count` in the `TaskList` class. 

```java
list.size(); //Returns the number of elements in the list
```

cxo05 commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,88 @@
+     */
+    public void printList() {
+        System.out.println("Here are the tasks in your list:");
+        for (int i = 0; i < count; i++) {
```
Instances of `count` can be replaced with `list.size()`.

cxo05 commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,88 @@
+    public void deleteTask(int index) {
+        this.list.remove(index);
+        System.out.println("Noted. I've removed this task:");
+        System.out.println(list.get(index));
```
ArrayList remove() returns the removed element

```java
Task removedTask = this.list.remove(index);
System.out.println(removedTask);
```

cxo05 commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,88 @@
+                return list.get(i);
+            }
+        }
+        return null;
```
You can just do 
```java
try {
  return list.get(index)
} catch (IndexOutOfBoundsException e) {
  // The element you are trying to access does not exist
  return null;
}
```

cxo05 commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,88 @@
+        return null;
+    }
+
+    public void findTask(String input, TaskList list) {
```
There is a special way to loop through ArrayLists in java using
```java
ArrayList<Task> foundTasks = new ArrayList<>();
for (Task task : this.list) {
  if (task.toString().contains(input)) {
    foundTasks.add(task);
  }
}

if (foundTasks.size() == 0) {
  // No tasks found
} else {
  for(int j = 0; j < foundTasks.size(); j++) {
    System.out.println((j + 1) + "." + foundTasks.get(j).toString());
  }
}
```

ravern commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,50 @@
+import java.io.FileNotFoundException;
+
+/**
+ * Starts the chatbot
```
Should there be a period/fullstop here? I noticed the same issue across your other JavaDoc comments.

https://se-education.org/guides/conventions/java/intermediate.html#comments

ravern commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,50 @@
+public class Duke {
+    private final Storage storage;
+    private final Ui ui;
+    private final TaskList list;
```
Great use of `final`!

ravern commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,50 @@
+     * Executes Duke
+     *
+     * @param args
+     * @throws FileNotFoundException
```
Should there be some descriptions for the `@param` and `@throws` here?

ravern commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,88 @@
+     * @param list Task is added into list
+     */
+
+    public static void parse(String input, TaskList list) {
```
Should there be an empty line between the JavaDoc comment and the method definition?

ravern commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,88 @@
+                String des = input.substring(5);
+                if (des.isBlank()) {
+                    throw new EmptyDescriptionException();
+                }
```
Great use of custom exception!

ravern commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,88 @@
+                }
+                String[] split = input.substring(6).split(" /from ");
+                String des = split[0];
+                String[] fromto = split[1].split(" /to ");
```
Perhaps `fromto` should be in camelCase i.e. `fromTo`?

https://se-education.org/guides/conventions/java/intermediate.html#naming

ravern commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,43 @@
+        if (file.exists()) {
+            try (Scanner fileScanner = new Scanner(file)) {
+                while (fileScanner.hasNextLine()) {
+                    String task = fileScanner.nextLine();
+                    Task dtask = Taskparser.parseTask(task); //convert format of tasks in file to list
```
Perhaps a better naming here would be `String taskString` and `Task task`?

ravern commented on src/main/java/duke/Taskparser.java:
```diff
@@ -0,0 +1,42 @@
+package duke;
+
+public class Taskparser {
```
Should this be in PascalCase i.e. `TaskParser`?

ravern commented on src/test/java/duke/TaskListTest.java:
```diff
@@ -0,0 +1,25 @@
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+public class TaskListTest {
+	@Test
+	public void addTest() {
+		TaskList list = new TaskList(new ArrayList<>());
+		list.addTask(new Todo("read book"));
+		assertEquals("[T][ ] read book",list.getTask(0).toString());
+	}
+
+	@Test
+	public void deleteTest() {
+		TaskList list = new TaskList(new ArrayList<>());
+		list.addTask(new Todo("read book"));
+		list.addTask(new Todo("study"));
+		list.deleteTask(0);
+		assertEquals("[T][ ] study",list.getTask(0).toString());
+	}
```
Is the indentation here correct?

ravern reviewed (commented):
Great work!

Minor note: Perhaps your code can be broken down further into different packages e.g. `task`, `errors`

saraozn committed:
fixed a few bugs

saraozn committed:
no message

saraozn committed:
resolved conflicts

saraozn committed:
fixed a few bugs

saraozn committed:
JavaFX

saraozn committed:
no message

saraozn committed:
feat: created GUI

saraozn committed:
Merge branch 'branch-Level-10'

saraozn committed:
resolved conflicts

saraozn committed:
Add header comments for more of the methods and classes

saraozn committed:
Taskparser.java: add assertions

saraozn committed:
Duke.java: Shorten run() method

Method is too long

Extract some of the implementation into Parser class

saraozn committed:
delete dead code

saraozn committed:
Merge pull request #2 from saraozn/branch-A-CodeQuality

Branch a code quality

saraozn committed:
Merge pull request #3 from saraozn/branch-A-Assertions

Taskparser.java: add assertions

saraozn committed:
Add method to edit description of task

Task has to be deleted and re-added to change the contents of the task

saraozn committed:
Merge branch 'branch-C-Update'

saraozn committed:
Fix display issues

saraozn committed:
Update user guide

saraozn committed:
update representative screenshot

saraozn committed:
update chatbot name

saraozn committed:
Merge branch 'branch-A-UserGuide'

saraozn committed:
add header comments

saraozn committed:
Edit findTask method to use ArrayList instead of List

List only allows a fixed maximum number of elements to be added.
ArrayList is more flexible in terms of maximum limit

saraozn committed:
add exit class

gui close automatically when user input bye

saraozn committed:
no message
